app/routers/case_information.py

```diff
@@ -33,9 +33,15 @@ async def read_all_cases(session: Session = Depends(get_session)) -> Sequence[Ca
 
 @router.post("/", tags=["cases"], response_model=CaseResponse)
 def create_case(request: CaseRequest, session: Session = Depends(get_session)):
-    data = request.translate()
-    case = Case(**data)
-    session.add(case)
-    session.commit()
-    session.refresh(case)
+    return request.create(session)
+
+
+@router.put("/{case_id}", tags=["cases"], response_model=CaseResponse)
+def update_case(
+    case_id: UUID, request: CaseRequest, session: Session = Depends(get_session)
+):
+    case = request.retrieve(session, case_id)
+    if case is None:
+        raise HTTPException(status_code=404, detail="Case not found")
+    request.update(case, session)
```

Comment:
When you update a Case all the sub models are re-created rather than updated.

Would it be possible to modify the original models rather than re-creating them? 

You could accept an Person/ Notes ID as an optional field and update it if it already exists.

Reply:
I have updated so that if a id is given then is updated.
This meant that we had to use a different create/update class for those nested relationships so on the update they accept an id.